### DISCUSSION TITLE
custom column type

### DISCUSSION THREAD

**Discussion by runnedrun** [AntlerVC/firetable#762] [Support / Q&A] [answered]
Is there currently or are there plans to add a flexible custom column type? For instance I store addresses in one my databases. They're stored as objects like so: 
{
  city: "Baltimore",
  zip: "21224",
  address: "1 main st"
  state: "MD"
}

In our current admin panel we display each component of the address in a separate column, to make editing easier and enforce formatting. We'd love to be able to do that or similar in rowy. Having a dedicated address field, of course, would be great, but as a stopgap, having an ability to define mapToDisplay function, and mapToWrite function, with code, would allow me to achieve what I want pretty easily.

**Comment by harinij** [comment, marked as answer]
Hi @runnedrun we have a flexible json column type that can be used for this usecase.

In this example you can have a json field called `address` and for the purpose of ease of editing on the Table UI, you can then individual columns as well. While creating these individual columns make sure to map the field key to the key of the json field. For example: a short text column to store city with field name `address.key`. Hope this solves your usecase.

<img width="482" alt="Screen Shot 2022-08-25 at 1 59 39 pm" src="https://user-images.githubusercontent.com/307298/186571890-8fca1f36-2011-4901-b847-488ded209a19.png">
